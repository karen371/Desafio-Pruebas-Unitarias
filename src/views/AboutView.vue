<!-- src/views/AboutView.vue -->
<template>
    <div>
      <h1>About</h1>
      <p>Esta es la pagina About</p>
    </div>
  </template>
  
  <script>
  export default {
    name: 'AboutView',
  };
  </script>