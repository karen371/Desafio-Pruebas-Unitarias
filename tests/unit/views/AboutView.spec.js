// tests/unit/views/AboutView.spec.js
import { shallowMount, mount } from '@vue/test-utils';
import { createRouter, createWebHistory } from 'vue-router'
import AboutView from '@/views/AboutView.vue';


/**Prueba para verificar la renderizacion de la vista */
describe('AboutView', () => {
  test('Probando la existencia del componente o vista AboutView', async () => {
      const router = createRouter({
          history: createWebHistory(),
          routes: [{
              path: '/about',
              name: 'AboutViewVue',
              component: AboutView
          }],
      });
      router.push('/about');
      await router.isReady();

      const wrapper = mount(AboutView, {
          global: {
              plugins: [router]
          }
      });
      expect(wrapper.findComponent(AboutView).exists()).toBe(true);
  });
});

/**TEST PARA LA VISTA ABOUT Snapshots*/
describe('Vista AboutView.vue', () => {
  test('ValidaciÃ³n de match con el snapshot', () => {
    const wrapper = shallowMount(AboutView);

    expect(wrapper.html()).toMatchSnapshot();
  });
});