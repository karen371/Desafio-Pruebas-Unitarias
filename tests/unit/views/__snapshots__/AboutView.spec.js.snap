// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vista AboutView.vue Validaci√≥n de match con el snapshot 1`] = `
<div>
  <h1>About</h1>
  <p>Esta es la pagina About</p>
</div>
`;
